jetty-security/src/main/java/org/eclipse/jetty/security/NoAuthentication.java

```diff
@@ -0,0 +1,58 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security;
+
+import javax.servlet.ServletRequest;
+
+import org.eclipse.jetty.security.authentication.LoginAuthenticator;
+import org.eclipse.jetty.server.Authentication;
+import org.eclipse.jetty.server.UserIdentity;
+
+/**
+ * NoAuthentication
+ *
+ * An Authentication indicating that a user has been previously, but is not currently logged in,
+ * but may be capable of logging in after a call to Request.login(String,String)
+ */
+public class NoAuthentication implements Authentication.NonAuthenticated
```

Comment:
I still prefer the name `LoggedOutAuthentication` as that describes the only way this authentication can be entered. It is unclear if `NoAuthentication` is because none has been provided or auth has not been attempted or has been attempted and failed?

Reply:
Done.